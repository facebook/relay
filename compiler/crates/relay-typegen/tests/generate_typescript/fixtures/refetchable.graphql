fragment FlowRefetchableFragment on Node
  @refetchable(queryName: "FlowRefetchableFragmentQuery") {
  id
  ... on User {
    name
  }
}
