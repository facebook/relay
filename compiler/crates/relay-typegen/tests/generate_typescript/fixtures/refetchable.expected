==================================== INPUT ====================================
fragment FlowRefetchableFragment on Node
  @refetchable(queryName: "FlowRefetchableFragmentQuery") {
  id
  ... on User {
    name
  }
}
==================================== OUTPUT ===================================
import { FragmentRefs } from "relay-runtime";
export type FlowRefetchableFragmentQueryVariables = {
  id: string;
};
export type FlowRefetchableFragmentQueryResponse = {
  readonly node: {
    readonly " $fragmentRefs": FragmentRefs<"FlowRefetchableFragment">;
  } | null;
};
export type FlowRefetchableFragmentQuery = {
  variables: FlowRefetchableFragmentQueryVariables;
  response: FlowRefetchableFragmentQueryResponse;
};
-------------------------------------------------------------------------------
import { FragmentRefs } from "relay-runtime";
export type FlowRefetchableFragment = {
  readonly id: string;
  readonly name?: string | null;
  readonly " $refType": "FlowRefetchableFragment";
};
export type FlowRefetchableFragment$data = FlowRefetchableFragment;
export type FlowRefetchableFragment$key = {
  readonly " $data"?: FlowRefetchableFragment$data;
  readonly " $fragmentRefs": FragmentRefs<"FlowRefetchableFragment">;
};
